src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -496,13 +496,50 @@ internal static IEnumerable<string> GetFrameworkAssembliesToTreatAsUserAssemblie
 			if (!File.Exists (acwPath))
 				return acw_map;
 			foreach (var s in File.ReadLines (acwPath)) {
-				var items = s.Split (';');
+				var items = s.Split (new char[] { ';' }, count: 2);
 				if (!acw_map.ContainsKey (items [0]))
 					acw_map.Add (items [0], items [1]);
 			}
 			return acw_map;
 		}
 
+		public static Dictionary<string, HashSet<string>> LoadCustomViewMapFile (IBuildEngine4 engine, string mapFile)
+		{
+			var cachedMap = (Dictionary<string, HashSet<string>>)engine?.GetRegisteredTaskObject (mapFile, RegisteredTaskObjectLifetime.Build);
+			if (cachedMap != null)
+				return cachedMap;
+			var map = new Dictionary<string, HashSet<string>> ();
+			if (!File.Exists (mapFile))
+				return map;
+			foreach (var s in File.ReadLines (mapFile)) {
+				var items = s.Split (new char [] { ';' }, count: 2);
+				var key = items [0];
+				var value = items [1];
+				HashSet<string> set;
+				if (!map.TryGetValue (key, out set))
+					map.Add (key, set = new HashSet<string> ());
+				set.Add (value);
+			}
+			return map;
+		}
+
+		public static void SaveCustomViewMapFile (IBuildEngine4 engine, string mapFile, Dictionary<string, HashSet<string>> map)
+		{
+			engine?.RegisterTaskObject (mapFile, map, RegisteredTaskObjectLifetime.Build, allowEarlyCollection: false);
+			var temp = Path.GetTempFileName ();
+			try {
+				using (var m = new StreamWriter (temp)) {
+					foreach (var i in map) {
+						foreach (var v in i.Value)
```

Comment:
Similarly, enumeration order of a `HashSet<string>` is undefined, so this should be `i.Value.OrderBy (f => f)` so that we can produce consistent files.